File: Location.java
package src;

public class Location
{
    public int x;
    public int y;

    public Location()
    {
        x = 0;
        y = 0;
    }

    public int get_x()
    {
        return x;
    }

    public int get_y()
    {
        return y;
    }

    public void print() 
    {
        System.err.println("(" + x + ", " + y + ")");
    }

    public String toString()
    {
        return "(" + x + ", " + y + ")";
    }

}

File: Location.c
#include "Location.h"
#include <stdio.h>
#include <stdlib.h>

Location* createLocation() {
    Location* newLocation = (Location*)malloc(sizeof(Location));
    if (newLocation == NULL) {
        fprintf(stderr, "Error: Memory allocation failed\n");
        exit(EXIT_FAILURE);
    }
    newLocation->x = 0;
    newLocation->y = 0;
    return newLocation;
}

int getX(Location* location) {
    return location->x;
}

int getY(Location* location) {
    return location->y;
}

void printLocation(Location* location) {
    printf("(%d, %d)\n", location->x, location->y);
}

char* locationToString(Location* location) {
    char* str = (char*)malloc(20 * sizeof(char));
    if (str == NULL) {
        fprintf(stderr, "Error: Memory allocation failed\n");
        exit(EXIT_FAILURE);
    }
    sprintf(str, "(%d, %d)", location->x, location->y);
    return str;
}

void destroyLocation(Location* location) {
    free(location);
}
